File: mobile/lib/models/student.dart
class Student {
  final String id;
  final String name;
  final String grade;
  final String classMatch; // "10th - A"
  final String email;
  final String status; // 'Active', 'Absent', 'Suspended'
  final String? avatarUrl;

  Student({
    required this.id,
    required this.name,
    required this.grade,
    required this.classMatch,
    required this.email,
    required this.status,
    this.avatarUrl,
  });
  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      grade: json['grade'] ?? '',
      classMatch: json['classMatch'] ?? '',
      status: json['status'] ?? 'Active',
      avatarUrl: json['avatar'],
    );
  }
}

File: web/src/backend/models/Student.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IStudent extends Document {
  name: string;
  email: string;
  grade: string;
  status: 'ACTIVE' | 'ABSENT' | 'SUSPENDED';
  avatar?: string;
  classId?: mongoose.Types.ObjectId;
  createdAt: Date;
  updatedAt: Date;
}

const StudentSchema = new Schema<IStudent>(
  {
    name: { type: String, required: true },
    email: { type: String, required: true, unique: true },
    grade: { type: String, required: true },
    status: { type: String, enum: ['ACTIVE', 'ABSENT', 'SUSPENDED'], default: 'ACTIVE' },
    avatar: { type: String },
    classId: { type: Schema.Types.ObjectId, ref: 'SchoolClass' },
  },
  { timestamps: true }
);

export default mongoose.models.Student || mongoose.model<IStudent>('Student', StudentSchema);
